### PR TITLE
feat(iOS): Add iOS Geo Category Maps and Search Developer Preview

### PR DESCRIPTION
**Pull request by ameter**
_Description of changes:_
Add iOS Geo Category Maps and Search Developer Preview

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by atierian — APPROVED** [review]
LGTM

File: src/fragments/lib/geo/ios/maps/10_install_adapter.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Amplify-MapLibre is an open source adapter that enables the popular MapLibre SDK to work seamlessly with Amplify Geo.
+
+1. To install the Amplify-MapLibre adapter to your application, open your project in Xcode and select **File > Add Packages...**
+
+1. Enter the amplify-ios-maplibre GitHub repo URL (`https://github.com/aws-amplify/amplify-ios-maplibre`) into the search bar and hit **Enter** Wait for the result to load. You'll see the repository rules for which version of amplify-ios-mapLibre you want Swift Package Manager to install.
```
Comment (siegerts) [inline src/fragments/lib/geo/ios/maps/10_install_adapter.mdx:5]:
```suggestion
1. Enter the amplify-ios-maplibre GitHub repo URL (`https://github.com/aws-amplify/amplify-ios-maplibre`) into the search bar and hit **Enter**. Wait for the result to load. You'll see the repository rules for which version of amplify-ios-mapLibre you want Swift Package Manager to install. 
```

File: src/fragments/lib/geo/ios/getting_started/20_cli_resources.mdx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+amplify add geo
+```
+
+The CLI will prompt configuration options for the Geo category for what type of capability you want to add and default or advanced settings. The `add` command automatically creates the backend configuration. Once all your configuration is complete run the following:
+
+```sh
+amplify push
+```
+
+Upon completion, amplifyconfiguration.json should be updated to reference provisioned backend geo resources. Note that this file should already be a part of your project if you followed the Project setup walkthrough.
```
Comment (siegerts) [inline src/fragments/lib/geo/ios/getting_started/20_cli_resources.mdx:37]: Maybe format `amplifyconfiguration.json` to highlight filename.

**Review by siegerts — APPROVED** [review]
Few comments but LGTM

File: src/fragments/lib/geo/ios/getting_started/20_cli_resources.mdx (hunk 2 above)
Comment (ameter) [reply src/fragments/lib/geo/ios/getting_started/20_cli_resources.mdx]: good catch.  done.

File: src/fragments/lib/geo/ios/getting_started/20_cli_resources.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Once that is complete, you can run the following command from your project's root folder to initialize Amplify in your repo:
+
+```sh
+amplify init
+```
+
+Now you are able to add a `geo` resource, such as map resources or a search index:
+
+```sh
+amplify add geo
```
Comment (royjit) [inline src/fragments/lib/geo/ios/getting_started/20_cli_resources.mdx:28]: Do we need to do `amplify add auth` also? Later in the step we are adding `try Amplify.add(plugin: AWSCognitoAuthPlugin())`

**Review by royjit — APPROVED** [review]
LGTM

**Pull request merged**